ghostscript: update to version 9.55.0, remove old versions

jmairboeck opened the pull request:
Fixes #6312, but untested, so not sure yet if it helps with #6284 (or the other issues with ghostscript in Lilypond)

Begasus commented:
Was looking into this also before I got side-tracked with poppler (again) :)
There is a typo patch floating around that other OS's use also: https://src.fedoraproject.org/rpms/ghostscript/blob/rawhide/f/ghostscript-9.55-fix-typo-in-gsicc-lcms2.patch
They also rm leptonica and tesseract in their recipes: https://github.com/archlinux/svntogit-packages/blob/packages/ghostscript/trunk/PKGBUILD#L32

jmairboeck commented:
The reason why it built before without that patch is because it actually still used the bundled version (lcms2mt). I have fixed that and included the patch.

I have also looked what to do with leptonica and tesseract: it seems that the configure script doesn't try to use the system version of these libs, presumably because they use modified versions. See https://ghostscript.com/blog/ocr.html for details. I have therefore retained these directories for now.

I also added openjpeg (to use the system library instead of the bundled one) and libiconv to dependencies to fix the build of a driver. The detection of dlopen in the configure script has also been fixed.

jmairboeck commented:
ghostscript 9.55.0 (and fig2dev) builds fine on 32 bit (using gcc11). It leads to the same crashes as on 64 bit ...
